src/shared/install.c

```diff
@@ -1708,6 +1708,92 @@ static int install_info_discover_and_check(
+        UnitType t1, t2;
+        int r;
+
+        const char *path_alias = strrchr(dst, '/');
+        if (path_alias) {
+                dir = dirname_malloc(dst);
+                if (!dir)
+                        return log_oom();
+
+                if (!endswith(dir, ".wants") && !endswith(dir, ".requires"))
+                        return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
+                                                 "Invalid path \"%s\" in alias.", dir);
+
+                /* That's the name we want to use for verification. */
+                dst = path_alias + 1;
+        }
+
+        un1 = unit_name_to_instance(i->name, &inst1);
+        if (un1 < 0)
+                return log_error_errno(un1, "Unit name %s is not valid: %m", i->name);
+
+        un2 = unit_name_to_instance(dst, &inst2);
+        if (un2 < 0)
+                return log_warning_errno(un2, "Unit name %s is not valid: %m", dst);
+
+        t1 = unit_name_to_type(i->name);
+        assert(t1 >= 0);
+        t2 = unit_name_to_type(dst);
+        assert(t2 >= 0);
+
+        if (t2 != t1)
+                return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
+                                         "Unit %s alias target %s has wrong type.", i->name, dst);
+
+        if (un1 == UNIT_NAME_PLAIN) {
+                if (un2 != UNIT_NAME_PLAIN)
+                        return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
+                                                 "Unit %s cannot be aliased by template %s",
+                                                 i->name, dst);
+
+        } else if (un1 == UNIT_NAME_TEMPLATE) {
+                bool ok =
+                        un2 == UNIT_NAME_TEMPLATE ||
+                        (un2 == UNIT_NAME_INSTANCE && streq_ptr(i->default_instance, inst2)) ||
+                        (path_alias && un2 != UNIT_NAME_PLAIN);
+
+                if (!ok)
+                        return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
+                                                 "Unit %s cannot be aliased by non-template %s",
+                                                 i->name, dst);
+
+        } else if (un1 == UNIT_NAME_INSTANCE) {
+                if (un2 == UNIT_NAME_PLAIN)
+                        return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
+                                                 "Unit %s cannot be aliased by plain unit %s",
+                                                 i->name, dst);
+
+                if (un2 == UNIT_NAME_TEMPLATE) {
+                        if (path_alias)
+                                return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
+                                                         "Unit %s cannot be %s as %s",
+                                                         i->name,
+                                                         endswith(dir, ".wants") ? "wanted" : "required",
+                                                         dst);
```

Comment:
this means that a template with `Alias=foot.target.wants/template@.service` can be enabled but the instances of the template can't be enabled ?

Reply:
In the version I'm about too push, neither is allowed.